tool2schema/schema.py

```diff
@@ -83,6 +83,14 @@ def __init__(self, func, **kwargs) -> None:
         functools.update_wrapper(self, func)
 
     def __call__(self, *args, **kwargs):
+
+        for key in kwargs:
+            if key in self.schema.parameter_schemas:
+                # Convert the JSON value to the type expected by the method
+                kwargs[key] = self.schema.parameter_schemas[key].parse_value(
+                    kwargs[key]
+                )
+
```

Comment:
There should be a way to also check `args` so you can also support the following test case:

```python
a, _, _, _ = function_custom_enum(CustomEnum.A.name, "", False, [])
assert a == CustomEnum.A
```

Unless you think this is something we shouldn't do?

Reply:
I did it this way because I assumed that when the function is invoked with data coming from the API we only use keyword arguments. But if you think it's useful in some case to parse args as well I will do it

Reply:
I think it will be good for completeness.

Reply:
Will do